packages/flutter_tools/lib/src/device.dart

```diff
@@ -378,6 +378,8 @@ abstract class DeviceLogReader {
 
   @override
   String toString() => name;
+
+  set pid(int value) { }
```

Comment:
This was done in case subclasses needed to add side effects when the pid is set, but it looks like that's unnecessary.  Moved this to a field in the superclass.